lib/pos_blockchain.go

```diff
@@ -530,6 +533,17 @@ func (bc *Blockchain) canCommitGrandparent(currentBlock *BlockNode) (_grandparen
 	return nil, false
 }
 
+func (bc *Blockchain) setBlockCommittedStatus(blockNode *BlockNode, status CommittedBlockStatus) {
+	bc.bestChainMap[*blockNode.Hash].CommittedStatus = status
+	bc.blockIndex[*blockNode.Hash].CommittedStatus = status
+	for _, node := range bc.bestChain {
+		if node.Hash.IsEqual(blockNode.Hash) {
+			node.CommittedStatus = status
+			break
+		}
+	}
```

Comment:
I'm not sure we need to iterate over the bestChain slice here. Assuming the bestChainMap has type map[BlockHash]*BlockNode (with value being a pointer to the BlockNode), then simply updating the status of the block node in the bestChainMap should the status of the underlying BlockNode value

Reply:
Yep - it *should* be using the same pointer.